src/integrators/wrapper_integrator.hpp

```diff
@@ -156,6 +157,64 @@ class MixedBilinearToNonlinearFormIntegrator : public mfem::NonlinearFormIntegra
   std::shared_ptr<mfem::ParFiniteElementSpace> trial_fes_;
 };
 
+/**
+ * @brief A class to convert NonlinearFormIntegrator to one where the input parameter undergoes a change of variables
+ */
+class SubstitutionNonlinearFormIntegrator : public mfem::NonlinearFormIntegrator {
```

Comment:
I think `TransformedNonlinearFormIntegrator` is more descriptive of what is going on (unless I'm missing something).